helix-rest/src/main/java/org/apache/helix/rest/server/service/InstanceServiceImpl.java

```diff
@@ -275,11 +275,18 @@ private StoppableCheck performCustomInstanceCheck(String clusterId, String insta
         _dataAccessor.getChildValues(_dataAccessor.keyBuilder().externalViews(), true);
     Map<String, StoppableCheck> instanceStoppableChecks = new HashMap<>();
     for (String instanceName : instances) {
-      List<String> unHealthyPartitions = InstanceValidationUtil
+      Map<String, List<String>> unHealthyPartitions = InstanceValidationUtil
           .perPartitionHealthCheck(externalViews, allPartitionsHealthOnLiveInstance, instanceName,
               _dataAccessor);
+
+      List <String> unHealthyPartitionsList = new ArrayList<>();
+      for (String partitionName : unHealthyPartitions.keySet()) {
+        unHealthyPartitionsList.add(partitionName + ":" + unHealthyPartitions.get(partitionName).stream()
+            .map(String::toUpperCase)
+            .collect(Collectors.joining(",")));
```

Comment:
So you do the concatenate with comma, if I understand it correctly. Then it could be problematic since the stoppable checks will be concatenated with comma as well, you will see it like this:

"instance 1: [failed reason1, CUSTOM_PARTITION:partition1, PARTITION_INITIAL_FAIL]xxxxx

Then it will be confused for user to understand which partition it fails, especially when json format read it into different order.

Reply:
Thanks. As discussed offline, I changed the format. Please have a look